### PR TITLE
test: expand unit and error-path coverage to 94 percent and raise gate to 90

### PR DESCRIPTION
**Pull request by nao1215**
## Summary

Raises combined unit and self-hosted E2E statement coverage from about 83 percent to 94 percent, and moves the octocov acceptable threshold from 80 percent to 90 percent.

The self-hosted atago E2E suite already contributes to the combined coverage number via `make coverage` (unit covdata merged with the coverage-instrumented binary's E2E covdata). This change concentrates on the statements that neither the unit tests nor the E2E specs reached, most of them error-propagation branches.

## What was added

New tests exercise error and edge paths across every layer:

- shell: usecase failures surfaced through the command handlers (`.header`, `.tables`, `.describe`, `.schema`, `.dump`), missing-table and no-tables reporting for `--inspect`, `--profile`, and `--compare`, `--compare-tables` argument validation, and direct tests for the `.cd` and `.ls` commands.
- filesql: closed-database returns and reachable error branches in the adapter, plus compression, parquet, ACH, and FedWire dump paths.
- memory and persistence: every repository method now has a "returns an error, does not panic, when the backing DB is closed" test that covers its transaction-begin error branch.
- config, interactor, domain/model: pure helpers (`hasReturningClause`, `mainStatementVerb`, table writers) and their edge cases.

## Coverage config

- `acceptable` raised from 80 percent to 90 percent.
- `cd.go` and `ls.go` removed from the exclude list. They are product code and are now directly tested, so excluding them no longer reflects reality.

## Notes

The remaining uncovered statements are overwhelmingly fault-injection-only branches: SQLite driver faults (scan/commit/rows.Err after a mid-operation failure), tablewriter render errors, `json.Marshal` failures, and OS syscall failures (`os.Getwd`, `MkdirTemp`, permission-denied walks). Reaching them would require broken-driver or broken-filesystem mocks that add little value, so the effort stops at the reachable set. No product bugs were found while writing these tests.

## Summary by CodeRabbit

* **Tests**
  * Expanded coverage across configuration, shell commands, import/export formatting, persistence/history, and in-memory database behavior.
  * Added assertions for error propagation and edge cases (e.g., write failures, invalid inputs, closed-database handling, cache behaviors, and compare/inspect/export error paths).

* **Chores**
  * Increased the minimum coverage requirement (80% → 90%) and adjusted coverage exclusions now that previously skipped paths are exercised.
  * Updated the changelog to document the increased statement coverage and added test scenarios.

### CODE REVIEW

File: shell/coverage_shell_test.go [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+		}
+		defer cleanup()
+		shell.argument.Output.FilePath = filepath.Join(t.TempDir(), "out.csv")
+
+		err = shell.runSQLFileToOutput(context.Background(), "SELECT 1 AS a;\nSELECT 2 AS b;")
+		if err == nil || !strings.Contains(err.Error(), "single result set") {
+			t.Fatalf("runSQLFileToOutput two-result error = %v, want single-result error", err)
+		}
+	})
+}
```
[the remark this answers is not in the document]
Comment (nao1215) [reply shell/coverage_shell_test.go:489]: Fixed. Removed t.Parallel() from this test and its three subtests so the config.Stderr swap in the "one result set" case can no longer race the sibling runSQLFileToOutput calls. Verified with go test -race.

File: shell/coverage_shell_test.go [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+	// We only require that it does not panic and returns cleanly; any suggestions
+	// carry the Excel-sheet description.
+	got := shell.getCompletions(context.Background(), ".import testdata/sample.xlsx --sheet=")
+	for _, s := range got {
+		if s.Description != "" && s.Description != msgExcelSheet {
+			continue
+		}
+	}
+	_ = got
+}
```
[the remark this answers is not in the document]
Comment (nao1215) [reply shell/coverage_shell_test.go:533]: Fixed. The loop now asserts with t.Errorf on any non-sheet suggestion instead of continuing and discarding the result.
